src/main.cpp

```diff
@@ -408,8 +409,21 @@ static void parse_env(BPFtrace& bpftrace)
       extra_flags.push_back(file);
     }
 
-    if (!clang.parse(driver.root.get(), bpftrace, extra_flags))
+    if (!clang.parse(driver.root.get(), bpftrace, extra_flags)) {
+      if (!found_kernel_headers) {
+        LOG(WARNING)
+            << "Could not find kernel headers in /lib/modules/"
```

Comment:
Nit: Maybe make a small utility function so we don't have to create this path twice e.g.
```
std::string get_default_kdir(const struct utsname &utsname)
{
  return std::string("/lib/modules/") + utsname.release
}
```

Then just pass that into `get_kernel_dirs`

Reply:
I considered that and am not sure... Just `/lib/modules/(uname -r)` isn't enough without source/build - nixos in particular has a `linuxHeaders` package that has no relation to the modules path and doesn't have a `build` or `source` component, if we want to override the directory at some point I think sharing now will mostly be undone so it's not really worth it
(unfortunately, nixpkgs' linuxHeaders doesn't have a kconfig.h as it's not directly related to the current kernel build, so it wasn't usable as is when I tried and I stopped looking this direction)

OTOH I agree it's a bit of a shame to have this twice, if you really prefer having such a helper I'll fix this tomorrow

Reply:
uh, actually I now (re-)notice KERNEL_HEADERS_DIR, so part of my message doesn't make sense (it's already overridable) -- however if that's set then get_kernel_dirs() cannot fail so this message would never be printed and the inconsistency doesn't really matter either.
It might make sense to move KERNEL_HEADERS_DIR in a new get default path helper and check if it exists though? In which case I guess the helper makes more sense, do we want to check if the hardcoded path exists and allows env vars to work with it?

Reply:
Ah yeah, I see what you're saying. How about this, we change the signature to something like:
```
bool get_kernel_dirs(
    const struct utsname &utsname, std::string& ksrc, std::string kobj)
```

The return bool will indicate if we failed to find the default kernel headers that you can then check if the parsing fails.

Reply:
This prototype makes sense, updated.
I've kept the cmake define first and without check to not change behaviour, but removed the #ifdef to at least make sure it builds